tests/cnf/ran/diskencryption/internal/stdin-matcher/stdinmatcher.go

```diff
@@ -0,0 +1,100 @@
+// Original code from https://github.com/greyerof/stdin-matcher
+
+package stdinmatcher
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"regexp"
+	"time"
+)
+
+// Matcher struc to hold regex to match for and the minimum of matches expected.
+type Matcher struct {
+	// Line regexp.
+	Regex *regexp.Regexp
+	// Num times (lines) that we're waiting the regex to match.
+	Times int
+}
+
+// WaitForAnyMatch waits for expected regex matches
+// if function found a match, the first match index is return as index in the match slice
+// If there was not match after timeout duration, timedout is set to true.
+func WaitForAnyMatch(input io.Reader, matches []Matcher, timeout time.Duration) (index int, timedout bool, err error) {
```

Comment:
In general the preference is to use [`wait.PollUntilContextTimeout`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollUntilContextTimeout) when possible since it does simplify things like the handling for errors and timeouts here

Reply:
I don't think we can use PollUntilContextTimeout here because we are not retrying, we are continuously streaming the console text and processing.